python/activity_stream/activity_stream.py

```diff
@@ -412,6 +496,39 @@ def load_data(self, sg_entity_dict):
         self._bundle.log_debug("Ask db manager to ask shotgun for updates...")
         self._data_manager.rescan()
         self._bundle.log_debug("...done")
+
+    def show_new_note_dialog(self, modal=True):
+        """
+        Shows a dialog that allows the user to input a new note.
+
+        :param modal:   Whether the dialog should be shown modally or not.
+        :type modal:    bool
+        """
+        note_dialog = NoteInputDialog(parent=self)
+        note_dialog.entity_created.connect(self._on_entity_created)
+        note_dialog.data_updated.connect(self.rescan)
+        note_dialog.set_bg_task_manager(self._task_manager)
+        note_dialog.set_current_entity(self._entity_type, self._entity_id)
+
+        if modal:
+            note_dialog.exec_()
+        else:
+            note_dialog.show()
+
+
+    def rescan(self, force_activity_stream_update=False):
+        """
+        Triggers a rescan of the current activity stream data.
+
+        :param force_activity_stream_update:    If True, will force a requery
```

Comment:
The way I'm using it is to force a recache of the stream data after there's been an update to the Shotgun data after I know the widget has already cached. In my case it's adding new file attachments to a Note entity, but there are other uses. If/When we expose the ability to edit fields on an entity (like a Note) from this widget (or a higher-level widget like the version details widget) we'll need a way to invalidate the sql lite cache for that entity.